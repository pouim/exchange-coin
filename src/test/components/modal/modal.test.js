import { shallow } from 'enzyme';
import Modal from 'src/components/ui/modal';

describe('Modal', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Modal debug />);

        expect(component).toMatchSnapshot();
    });

    it('should render container and two buttons', () => {
        const component = shallow(<Modal visible={true} />);
        const cancelButton = component.find('[data-test="modal-cancel"]');
        const confirmButton = component.find('[data-test="modal-confirm"]');
        const containerButton = component.find('[data-test="modal-container"]');

        expect(cancelButton).toHaveLength(1);
        expect(confirmButton).toHaveLength(1);
        expect(containerButton).toHaveLength(1);
    });
});
