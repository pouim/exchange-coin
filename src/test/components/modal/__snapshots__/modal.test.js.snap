// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render c 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render container and two buttons 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly with no props 1`] = `ShallowWrapper {}`;
